### ISSUE TITLE
baseUrl is used to create reference type path in declaration file

### ISSUE THREAD

**Issue by sheetalkamat** [microsoft/TypeScript#39695]
`npm run build` is definitely bug and #39645 fixes that.

` npm run build-baseUrl ` well that raises questions. #32878 started preferring `types` over `reference paths` to make it more portable. But our resolve type reference name doesnt use baseUrl so this is kind of questionable.
Other way to look at it would be when file is referenced throught only referece path from root file tree (or relative imports) prefer relative path instead of portable path.

_Originally posted by @sheetalkamat in https://github.com/microsoft/TypeScript/issues/37928#issuecomment-660350588_

**Comment by sheetalkamat** [comment]
Note that this is not regression. #32878 introduced this which is around for sometime now.

**Comment by sheetalkamat** [comment]
This needs more work and time to bake in so we can emit correct paths and falls into type reference resolution and declaration emit to match those expectations and hence kind of dependent on #39708

**Comment by Griffork** [comment]
This seems related to #34749 ? Presence of the baseUrl is breaking compiles in my project because `///reference path` gets converted to an invalid `///reference types`.

**Comment by sheetalkamat** [comment]
Duplicate of https://github.com/microsoft/TypeScript/issues/34749
